tests/test_execution_compute_plan.py

```diff
@@ -150,46 +143,170 @@ def test_compute_plan_single_session_failure(global_execution_env):
     spec = factory.create_algo(py_script=sbt.factory.INVALID_ALGO_SCRIPT)
     algo = session.add_algo(spec)
 
-    cp_spec = factory.create_compute_plan(algo=algo, objective=objective)
+    cp_spec = factory.create_compute_plan(objective=objective)
 
     traintuple_spec_1 = cp_spec.add_traintuple(
+        algo=algo,
         dataset=dataset,
         data_samples=[data_sample_1]
     )
     cp_spec.add_testtuple(traintuple_spec_1)
 
     traintuple_spec_2 = cp_spec.add_traintuple(
+        algo=algo,
         dataset=dataset,
         data_samples=[data_sample_2],
-        traintuple_specs=[traintuple_spec_1]
+        in_models_tuples=[traintuple_spec_1]
     )
     cp_spec.add_testtuple(traintuple_spec_2)
 
     traintuple_spec_3 = cp_spec.add_traintuple(
+        algo=algo,
         dataset=dataset,
         data_samples=[data_sample_3],
-        traintuple_specs=[traintuple_spec_2]
+        in_models_tuples=[traintuple_spec_2]
     )
     cp_spec.add_testtuple(traintuple_spec_3)
 
     # Submit compute plan and wait for it to complete
-    cp = session.add_compute_plan(cp_spec)
+    cp_created = session.add_compute_plan(cp_spec)
+    cp = cp_created.future().wait()
 
-    traintuples = [
-        session.get_traintuple(key).future().wait(raises=False)
-        for key in cp.traintuple_keys
-    ]
-
-    testtuples = [
-        session.get_testtuple(key).future().wait(raises=False)
-        for key in cp.testtuple_keys
-    ]
+    traintuples = cp.list_traintuples(session)
+    testtuples = cp.list_testtuples(session)
 
     # All the train/test tuples should be marked as failed
     for t in traintuples + testtuples:
         assert t.status == 'failed'
 
-    compute_plan = session.get_compute_plan(cp.compute_plan_id)
-    assert cp.compute_plan_id == compute_plan.compute_plan_id
-    assert set(cp.traintuple_keys) == set(compute_plan.traintuples)
-    assert set(cp.testtuple_keys) == set(compute_plan.testtuples)
+    assert cp_created.compute_plan_id == cp.compute_plan_id
+    assert set(cp_created.traintuple_keys) == set(cp.traintuples)
+    assert set(cp_created.testtuple_keys) == set(cp.testtuples)
+
+
+def test_compute_plan_aggregate_composite_traintuples(factory, session_1, session_2):
```

Comment:
We could use the latest `execution_env` fixture.

Reply:
👍